### PR TITLE
Update rust-rocksdb

### PR DESCRIPTION
**Pull request by yiwu-arbug**
## What have you changed? (mandatory)
Update rust-rocksdb to cherry-pick the following fixes:
https://github.com/pingcap/rust-rocksdb/pull/311 Fix create_cf not passinig titan specific options to titan
https://github.com/pingcap/rust-rocksdb/pull/312 Fix Titan portable build
https://github.com/pingcap/rust-rocksdb/pull/315 Fix libtitan_sys sse flag

## What are the type of the changes? (mandatory)
Bug fix

## How has this PR been tested? (mandatory)
CI

## Does this PR affect documentation (docs) or release note? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

**Comment by yiwu-arbug** [comment]
Need to fix a typo before merge: https://github.com/pingcap/rust-rocksdb/pull/312#discussion_r300113842

**Comment by yiwu-arbug** [comment]
@Connor1996 @overvenus mind take another look? Thanks.

File: Cargo.lock [hunk 1]
```diff
@@ -2906,7 +2906,7 @@ dependencies = [
 [[package]]
 name = "zstd-sys"
 version = "1.4.10+zstd.1.4.0"
-source = "git+https://github.com/gyscos/zstd-rs.git#d09a3a06e2802e92f011bb362fa2b4c264ed6f41"
+source = "git+https://github.com/gyscos/zstd-rs.git#f5d0cddc58a1b164e7312164465d11bc701af83e"
```
Comment (overvenus) [inline Cargo.lock]: Is this change required?

Comment (yiwu-arbug) [reply Cargo.lock]: I don't think so. But this is done by `cargo update -p rocksdb`.

**Comment by Connor1996** [comment]
please also update rocksdb on master

**Comment by yiwu-arbug** [comment]
will do!
